### ISSUE TITLE
[BUG] How to compile out of the box?

### ISSUE THREAD

**Issue by KweezyCode** [Elytrium/LimboAPI#111]
```
17:55:13: Executing 'clean build'...

Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs/4.7.0/spotbugs-4.7.0.pom, took 136 ms (6,67 kB)
Download https://repo.maven.apache.org/maven2/org/junit/junit-bom/5.8.2/junit-bom-5.8.2.module, took 29 ms (6,96 kB)
Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs/4.7.0/spotbugs-4.7.0.module, took 31 ms (10,27 kB)
Download https://repo.maven.apache.org/maven2/jaxen/jaxen/1.2.0/jaxen-1.2.0.pom, took 33 ms (14,84 kB)
Download https://repo.maven.apache.org/maven2/com/google/code/gson/gson/2.9.0/gson-2.9.0.pom, took 29 ms (8,06 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.3/asm-9.3.pom, took 87 ms (2,37 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-analysis/9.3/asm-analysis-9.3.pom, took 91 ms (2,62 kB)
Download https://repo.maven.apache.org/maven2/net/sf/saxon/Saxon-HE/11.3/Saxon-HE-11.3.pom, took 93 ms (4,21 kB)
Download https://repo.maven.apache.org/maven2/com/google/code/gson/gson-parent/2.9.0/gson-parent-2.9.0.pom, took 28 ms (4,51 kB)
Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs-annotations/4.7.0/spotbugs-annotations-4.7.0.pom, took 27 ms (3,46 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-commons/9.3/asm-commons-9.3.pom, took 30 ms (2,96 kB)
Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs-annotations/4.7.0/spotbugs-annotations-4.7.0.module, took 32 ms (5,03 kB)
Download https://repo.maven.apache.org/maven2/org/dom4j/dom4j/2.1.3/dom4j-2.1.3.pom, took 28 ms (2,76 kB)
Download https://repo.maven.apache.org/maven2/org/apache/commons/commons-text/1.9/commons-text-1.9.pom, took 33 ms (16,86 kB)
Download https://repo.maven.apache.org/maven2/org/dom4j/dom4j/2.1.3/dom4j-2.1.3.module, took 30 ms (2,86 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-tree/9.3/asm-tree-9.3.pom, took 31 ms (2,59 kB)
Download https://repo.maven.apache.org/maven2/org/apache/commons/commons-parent/51/commons-parent-51.pom, took 62 ms (77,94 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-util/9.3/asm-util-9.3.pom, took 28 ms (2,94 kB)
Download https://repo.maven.apache.org/maven2/org/apache/bcel/bcel/6.5.0/bcel-6.5.0.pom, took 35 ms (21,02 kB)
Download https://repo.maven.apache.org/maven2/org/apache/commons/commons-parent/50/commons-parent-50.pom, took 64 ms (75,62 kB)
Download https://repo.maven.apache.org/maven2/org/xmlresolver/xmlresolver/4.2.0/xmlresolver-4.2.0.pom, took 29 ms (2,25 kB)
Download https://repo.maven.apache.org/maven2/org/xmlresolver/xmlresolver/4.2.0/xmlresolver-4.2.0.module, took 32 ms (4,57 kB)
Download https://repo.maven.apache.org/maven2/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.pom, took 29 ms (2,66 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.3/asm-9.3.jar, took 54 ms (122,18 kB)
Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs-annotations/4.7.0/spotbugs-annotations-4.7.0.jar, took 28 ms (15,2 kB)
Download https://repo.maven.apache.org/maven2/com/google/code/gson/gson/2.9.0/gson-2.9.0.jar, took 68 ms (249,28 kB)
Download https://repo.maven.apache.org/maven2/org/apache/commons/commons-text/1.9/commons-text-1.9.jar, took 46 ms (216,21 kB)
Download https://repo.maven.apache.org/maven2/org/dom4j/dom4j/2.1.3/dom4j-2.1.3.jar, took 139 ms (323,63 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-analysis/9.3/asm-analysis-9.3.jar, took 32 ms (34,28 kB)
Download https://repo.maven.apache.org/maven2/net/jcip/jcip-annotations/1.0/jcip-annotations-1.0.jar, took 44 ms (2,25 kB)
Download https://repo.maven.apache.org/maven2/jaxen/jaxen/1.2.0/jaxen-1.2.0.jar, took 109 ms (232,46 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-tree/9.3/asm-tree-9.3.jar, took 53 ms (52,67 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-util/9.3/asm-util-9.3.jar, took 48 ms (85,68 kB)
Download https://repo.maven.apache.org/maven2/org/ow2/asm/asm-commons/9.3/asm-commons-9.3.jar, took 41 ms (72,72 kB)
Download https://repo.maven.apache.org/maven2/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar, took 120 ms (220,54 kB)
Download https://repo.maven.apache.org/maven2/org/apache/bcel/bcel/6.5.0/bcel-6.5.0.jar, took 414 ms (695,05 kB)
Download https://repo.maven.apache.org/maven2/org/xmlresolver/xmlresolver/4.2.0/xmlresolver-4.2.0.jar, took 606 ms (150,26 kB)
Download https://repo.maven.apache.org/maven2/org/xmlresolver/xmlresolver/4.2.0/xmlresolver-4.2.0-data.jar, took 921 ms (1,03 MB)
Download https://repo.maven.apache.org/maven2/com/github/spotbugs/spotbugs/4.7.0/spotbugs-4.7.0.jar, took 1 s 332 ms (3,54 MB)
Download https://repo.maven.apache.org/maven2/net/sf/saxon/Saxon-HE/11.3/Saxon-HE-11.3.jar, took 1 s 468 ms (5,11 MB)
> Task :clean UP-TO-DATE
> Task :api:clean UP-TO-DATE
> Task :plugin:clean UP-TO-DATE
> Task :checkLicenseMain NO-SOURCE
> Task :checkLicenseTest NO-SOURCE
> Task :checkLicenses UP-TO-DATE
> Task :api:checkLicenseTest NO-SOURCE
> Task :plugin:checkLicenseMain
> Task :plugin:checkLicenseTest NO-SOURCE
> Task :plugin:checkLicenses
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :assemble
> Task :checkstyleMain NO-SOURCE
> Task :compileTestJava NO-SOURCE
> Task :processTestResources NO-SOURCE
> Task :testClasses UP-TO-DATE
> Task :checkstyleTest NO-SOURCE
> Task :spotbugsMain NO-SOURCE
> Task :spotbugsTest NO-SOURCE
> Task :test NO-SOURCE
> Task :check UP-TO-DATE
> Task :build
> Task :api:generateTemplates
> Task :api:compileJava
> Task :api:processResources NO-SOURCE
> Task :api:classes
> Task :api:jar

> Task :api:javadoc
warning: URL https://jd.adventure.kyori.net/nbt/4.11.0/element-list was redirected to https://jd.advntr.dev/nbt/4.11.0/element-list -- Update the command-line options to suppress this warning.
1 warning

> Task :api:javadocJar
> Task :api:sourcesJar
> Task :api:assemble
> Task :api:checkLicenseMain
> Task :api:checkLicenses
Download https://repo.maven.apache.org/maven2/com/puppycrawl/tools/checkstyle/10.3/checkstyle-10.3.pom, took 372 ms (152,36 kB)
Download https://repo.maven.apache.org/maven2/info/picocli/picocli/4.6.3/picocli-4.6.3.pom, took 29 ms (1,38 kB)
Download https://repo.maven.apache.org/maven2/org/antlr/antlr4-runtime/4.10.1/antlr4-runtime-4.10.1.pom, took 106 ms (3,47 kB)
Download https://repo.maven.apache.org/maven2/org/reflections/reflections/0.10.2/reflections-0.10.2.pom, took 78 ms (9,32 kB)
Download https://repo.maven.apache.org/maven2/org/antlr/antlr4-master/4.10.1/antlr4-master-4.10.1.pom, took 30 ms (4,38 kB)
Download https://repo.maven.apache.org/maven2/org/checkerframework/checker-qual/3.12.0/checker-qual-3.12.0.module, took 30 ms (3,55 kB)
Download https://repo.maven.apache.org/maven2/org/javassist/javassist/3.28.0-GA/javassist-3.28.0-GA.pom, took 30 ms (10,81 kB)
Download https://repo.maven.apache.org/maven2/info/picocli/picocli/4.6.3/picocli-4.6.3.jar, took 124 ms (405,95 kB)
Download https://repo.maven.apache.org/maven2/org/antlr/antlr4-runtime/4.10.1/antlr4-runtime-4.10.1.jar, took 126 ms (322,14 kB)
Download https://repo.maven.apache.org/maven2/org/javassist/javassist/3.28.0-GA/javassist-3.28.0-GA.jar, took 195 ms (851,53 kB)
Download https://repo.maven.apache.org/maven2/org/reflections/reflections/0.10.2/reflections-0.10.2.jar, took 32 ms (130,4 kB)
Download https://repo.maven.apache.org/maven2/com/puppycrawl/tools/checkstyle/10.3/checkstyle-10.3.jar, took 284 ms (2,02 MB)
> Task :api:checkstyleMain
> Task :api:compileTestJava NO-SOURCE
> Task :api:processTestResources NO-SOURCE
> Task :api:testClasses UP-TO-DATE
> Task :api:checkstyleTest NO-SOURCE
Download https://repo.maven.apache.org/maven2/org/slf4j/slf4j-simple/1.8.0-beta4/slf4j-simple-1.8.0-beta4.pom, took 94 ms (1,11 kB)
Download https://repo.maven.apache.org/maven2/org/slf4j/slf4j-simple/1.8.0-beta4/slf4j-simple-1.8.0-beta4.jar, took 32 ms (15,42 kB)
> Task :api:spotbugsTest NO-SOURCE
> Task :api:test NO-SOURCE

> Task :plugin:compileJava
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/LimboAPI.java:438: warning: [removal] registerPacket(PacketDirection,Class<?>,Supplier<?>,PacketMapping[]) in LimboFactory has been deprecated and marked for removal
  public void registerPacket(PacketDirection direction, Class<?> packetClass, Supplier<?> packetSupplier, PacketMapping[] packetMappings) {
              ^
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/injection/login/LoginTasksQueue.java:237: error: cannot find symbol
              connection.setSessionHandler(
                        ^
  symbol:   method setSessionHandler(InitialConnectSessionHandler)
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/injection/login/LoginListener.java:133: error: cannot find symbol
      Object handler = connection.getSessionHandler();
                                 ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/injection/login/LoginListener.java:227: error: cannot find symbol
      if (!(connection.getSessionHandler() instanceof ClientPlaySessionHandler)) {
                      ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/injection/login/LoginListener.java:231: error: cannot find symbol
          connection.setSessionHandler(playHandler);
                    ^
  symbol:   method setSessionHandler(ClientPlaySessionHandler)
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboImpl.java:385: error: cannot find symbol
          connection.getSessionHandler(),
                    ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboImpl.java:389: error: cannot find symbol
      connection.setSessionHandler(sessionHandler);
                ^
  symbol:   method setSessionHandler(LimboSessionHandlerImpl)
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboPlayerImpl.java:264: error: cannot find symbol
    LimboSessionHandlerImpl handler = (LimboSessionHandlerImpl) this.connection.getSessionHandler();
                                                                               ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboPlayerImpl.java:284: error: cannot find symbol
    LimboSessionHandlerImpl handler = (LimboSessionHandlerImpl) this.connection.getSessionHandler();
                                                                               ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboPlayerImpl.java:357: error: cannot find symbol
    LimboSessionHandlerImpl handler = (LimboSessionHandlerImpl) this.connection.getSessionHandler();
                                                                               ^
  symbol:   method getSessionHandler()
  location: variable connection of type MinecraftConnection
/home/kweezy/Desktop/LimboAPI/plugin/src/main/java/net/elytrium/limboapi/server/LimboSessionHandlerImpl.java:298: error: cannot find symbol
      connection.eventLoop().execute(() -> connection.setSessionHandler(this.originalHandler));
                                                     ^
                                                    ```

**Comment by DHCPCD9** [comment]
Use older velocity api version.

**Comment by KweezyCode** [comment]
> Use older velocity api version.

then it should be fixed in repo

**Comment by Dreeam-qwq** [comment]
> > Use older velocity api version.
> 
> then it should be fixed in repo

You can use “1.1.17” branch instead of master branch for now, then you can build it successfully

**Comment by KweezyCode** [comment]
> > > Use older velocity api version.
> > 
> > 
> > then it should be fixed in repo
> 
> You can use “1.1.17” branch instead of master branch for now, then you can build it successfully

i literally patched it myself then realized it was already patched in 1.1.17 xD anyway thx, it is compiling fine
